### PR TITLE
Prep for web console login

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@deads2k review

File: pkg/assets/handlers.go [hunk 1]
```diff
@@ -104,7 +108,7 @@ func HTML5ModeHandler(h http.Handler) http.Handler {
 	})
 }
 
-func GeneratedConfigHandler(osAddr string, osPrefix string, k8sAddr string, k8sPrefix string, h http.Handler) http.Handler {
+func GeneratedConfigHandler(osAddr, osPrefix, k8sAddr, k8sPrefix, oauthAuthURL, oauthClientID string, h http.Handler) http.Handler {
```
Comment (deads2k) [inline pkg/assets/handlers.go]: Got a few arguments, eh?  Is this a one off or do these get passed around elsewhere?

File: pkg/cmd/server/origin/auth.go [hunk 2]
```diff
@@ -159,17 +275,32 @@ func (c *AuthConfig) NewOpenShiftOAuthClientConfig(client *oauthapi.Client) *osi
...
+		AuthorizeUrl:             OpenShiftOAuthAuthorizeURL(c.MasterPublicAddr),
+		TokenUrl:                 OpenShiftOAuthTokenURL(c.MasterPublicAddr),
 		Scope:                    "",
 	}
 	return config
 }
 
-func CreateOrUpdateDefaultOAuthClients(masterPublicAddr string, clientRegistry oauthclient.Registry) {
+func OpenShiftOAuthAuthorizeURL(masterAddr string) string {
+	return masterAddr + OpenShiftOAuthAPIPrefix + osinserver.AuthorizePath
```
Comment (deads2k) [inline pkg/cmd/server/origin/auth.go]: I know this is pre-existing from me, but I think we're supposed to use path.join.

File: pkg/cmd/server/start.go [hunk 3]
```diff
@@ -398,12 +403,36 @@ func start(cfg *config, args []string) error {
...
-			MasterRoots:      roots,
-			SessionSecrets:   []string{"secret"},
-			EtcdHelper:       etcdHelper,
+			MasterAddr:           cfg.MasterAddr.URL.String(),
+			MasterPublicAddr:     masterPublicAddr.URL.String(),
+			AssetPublicAddresses: assetPublicAddresses,
+			MasterRoots:          roots,
+			EtcdHelper:           etcdHelper,
+
+			AuthRequestHandlers: origin.ParseAuthRequestHandlerTypes(env("ORIGIN_OAUTH_REQUEST_HANDLERS", "session,basicauth")),
```
Comment (deads2k) [inline pkg/cmd/server/start.go]: Use the enumerated values for specifying defaults.

File: pkg/oauth/server/osinserver/osinserver.go [hunk 4]
```diff
@@ -32,9 +38,9 @@ func (s *Server) Install(mux Mux, paths ...string) {
 	for _, prefix := range paths {
 		prefix = strings.TrimRight(prefix, "/")
 
-		mux.HandleFunc(prefix+"/authorize", s.handleAuthorize)
-		mux.HandleFunc(prefix+"/token", s.handleToken)
-		mux.HandleFunc(prefix+"/info", s.handleInfo)
+		mux.HandleFunc(prefix+AuthorizePath, s.handleAuthorize)
```
Comment (deads2k) [inline pkg/oauth/server/osinserver/osinserver.go]: path.join

**Comment by deads2k** [comment]
Minor comments.  I still think we need a better config story for authentication, but this is definitely an improvement.

**Comment by liggitt** [comment]
@deads2k comments addressed

**Comment by deads2k** [comment]
lgtm

**Comment by liggitt** [comment]
@sosiouxme review last commit?

**Comment by sosiouxme** [comment]
aside from go evidently lacking much facility for common array/hash operations, looks good to me :+1:

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
fixed unstable sort order, re[merge]

**Pull request merged**